### PR TITLE
add getting result image by find by image

### PR DESCRIPTION
**Pull request by KazuCocoa**
closes https://github.com/appium/appium/issues/12506

## What this will do
- Store the result of find by image in `ImageElement`
    - It is available only when `getMatchedImageResult` is `true` via settings API
- Users can get the data via `getAttribute` API as `visual`
    - e.g. Ruby: `element.visual`, Python: `element.getAttribute('visual')` etc

## example in Ruby

```ruby
> @driver.update_settings({ fixImageTemplateScale: true, getMatchedImageResult: false })=> nil
> @core.wait { @@driver.find_element_by_image AppiumLibCoreTest.path_of('test/functional/data/test_ios_button.png') }
=> #<Selenium::WebDriver::Element:0x7144c4bf699ffe78 id="appium-image-element-758743a6-7c14-4a90-932c-be7d247faef8">
> image_element.visual
=> nil
> @driver.update_settings({ fixImageTemplateScale: true, getMatchedImageResult: true })=> nil
> image_element = @@core.wait { @@driver.find_element_by_image AppiumLibCoreTest.path_of('test/functional/data/test_ios_button.png') }
=> #<Selenium::WebDriver::Element:0x..f814fd569a6a19222 id="appium-image-element-efb92580-e72a-4d96-9a61-4ea45137cd27">
> v = image_element.visual
=> "iVBORw0KGgoAAAANSUhEUgAAAXcAAAKbCAIAAACBzF3NAAAgAElEQVR4AezBC5SP1f4A/O/e+7n9rnM1YxgOE5FQ+QspoRDFTG6FiKKYRhqGSaHkLk0qJKEc5VqoSTdJF+pBShlyN+aMGcx9frfnuvd+1/tb76zVWvWuznSylnM8nw969NFHweFwOK4YtG/fPnA4HI4rBu3fvx8cDofjikH79u0Dh8PhuGKQqqrwR2zbBgBCiCAIUAchZNs2/FW6rjPGAMDv99u2DX+VbdsIIYyxIAiGYSCEOOeMMYwxIQRjzDnXdd3tdiOEDMMQBAGuJZRSqCPLMgBwziml8BuEEF3XEUKKoti2DX9EFMVIJMIYkySJUooQwhgjhACAc84Y41HwG7IsM8Y0TXO5XPAbnHPGGOccIQR1LMuSZRkhZBiGLMvwG5xzAMAYc87BUX+UUhSFMdY0Df4qWZYjkUhMTE..."
> image_element.unsupported_attribute
Selenium::WebDriver::Error::UnknownMethodError: Method has not yet been implemented
from NotYetImplementedError: Method has not yet been implemented
```

----

Will add this debug feature in http://appium.io/docs/en/advanced-concepts/image-elements/

### CODE REVIEW

File: lib/basedriver/image-element.js [hunk 1]
```diff
@@ -227,6 +238,15 @@ class ImageElement {
         return imgEl.location;
       case 'getElementRect':
         return imgEl.rect;
+      case 'getAttribute':
+        // /wd/hub/session/:sessionId/element/:elementId/attribute/:name
+        // /wd/hub/session/:sessionId/element/:elementId/attribute/visual should retun the visual data
+        // e.g. ["content-desc","appium-image-element-xxxxx","xxxxx"], ["visual","appium-image-element-xxxxx","xxxxx"]
+        if (args[0] === 'visual') {
+          return imgEl.matchedImage;
+        } else {
```
Comment (jlipps) [inline lib/basedriver/image-element.js]: pretty sure this else block is not necessary (without a `break` or `return`, the `default` case will also be executed)

Comment (KazuCocoa) [reply lib/basedriver/image-element.js]: Yes, but the lint claimed about this point.
I compared adding an ignoring the lint message vs adding this else block.

Which is safer in JS world? (I thought following the lint was safer for now.)

File: lib/basedriver/commands/find.js [hunk 2]
```diff
@@ -238,8 +240,10 @@ helpers.findByImage = async function findByImage (b64Template, {
         fixImageTemplateScale, ...scale
       });
 
-      rect = (await this.compareImages(MATCH_TEMPLATE_MODE, b64Screenshot,
-        b64Template, {threshold})).rect;
+      const comparedImage = await this.compareImages(MATCH_TEMPLATE_MODE, b64Screenshot, b64Template, {threshold, visualize});
```
Comment (mykola-mokhnach) [inline lib/basedriver/commands/find.js]: `const {rect, b64Matched} = ...`

**Comment by atimofeev86** [comment]
what about c#?

**Pull request merged**